### PR TITLE
Hoist some pures

### PR DESCRIPTION
**Pull request by nickdrozd**

### CODE REVIEW

File: src/Core/Core.idr [hunk 1]
```diff
@@ -449,9 +449,9 @@ export
 wrapError : (Error -> Error) -> Core a -> Core a
 wrapError fe (MkCore prog)
     = MkCore (prog >>=
-                 (\x => case x of
-                             Left err => pure (Left (fe err))
-                             Right val => pure (Right val)))
+                 (\x => pure $ case x of
+                             Left err => Left (fe err)
+                             Right val => Right val))
```
Comment (buzden) [inline src/Core/Core.idr:454]: You can go more radical:
```idris
wrapError fe (MkCore prog)      
    = MkCore $ prog <&>      
                 \case                  
                   Left err  => Left (fe err)                      
                   Right val => Right val
```
or better even more:
```idris
wrapError fe (MkCore prog)      
    = MkCore $ prog <&> mapFst fe
```

Comment (fabianhjr) [reply src/Core/Core.idr:454]: :+1: nice

Comment (nickdrozd) [reply src/Core/Core.idr:454]: What does that mean? I don't understand the `<&>` notation (or `<$>`).

Comment (fabianhjr) [reply src/Core/Core.idr:454]: Flipped version of `<$>`: https://github.com/idris-lang/Idris2/blob/a28bc655440e963591aaa7cfc47adefc4e785452/libs/prelude/Prelude/Interfaces.idr#L87-L93

Comment (fabianhjr) [reply src/Core/Core.idr:454]: And `f <$> traversable` is `map f traversable`, so `traversable <&> f` is also `map f traversable`.

Comment (nickdrozd) [reply src/Core/Core.idr:454]: I see. I'll take care of that in another PR.

**Pull request merged**